File: app/src/main/java/com/dh/testproject/activity/viewpager/card/Card.java
package com.dh.testproject.activity.viewpager.card;

import android.text.BidiFormatter;
import android.util.Log;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.Iterator;
import java.util.List;
import java.util.Set;

public class Card {
    private static final String ARGS_BUNDLE = Card.class.getSimpleName() + ":Bundle";
    private String suit;
    private String value;

    public Card(String suit, String value) {
        this.suit = suit;
        this.value = value;
    }


    private static Set<String> SUITS = new HashSet<String>() {
        {
            add("♣" /* clubs*/);
            add("♦" /* diamonds*/);
            add("♥" /* hearts*/);
            add("♠" /*spades*/);
        }
    };
    private static Set<String> VALUES = new HashSet<String>() {
        {
            add("2");
            add("3");
            add("4");
            add("5");
            add("6");
            add("7");
            add("8");
            add("9");
            add("10");
            add("J");
            add("Q");
            add("K");
            add("A");
        }
    };

    public static List<Card> getDECK() {
        List<Card> list = new ArrayList<>();
        Iterator<String> iterator = SUITS.iterator();
        for (String suit : SUITS) {
            Log.e(ARGS_BUNDLE, "suit: " + suit);
            for (String value : VALUES) {
                Log.e(ARGS_BUNDLE, "value: " + value);
                list.add(new Card(suit,value));
            }
        }
        return list;
    }

    @NonNull
    @Override
    public String toString() {
        BidiFormatter bidiFormatter = BidiFormatter.getInstance();
        if (bidiFormatter.isRtlContext()) {
            return bidiFormatter.unicodeWrap(suit + " " + value);
        }else {
            return bidiFormatter.unicodeWrap(value + " " + suit);
        }
    }
}

File: app/src/main/java/com/dh/testproject/activity/viewpager/Card.kt
package com.dh.testproject.activity.viewpager

import android.os.Bundle
import androidx.core.text.BidiFormatter

class Card private constructor(val suit: String, val value: String) {

    val cornerLabel: String
        get() = "$suit\n$value"

    /** Use in conjunction with [Card.fromBundle]  */
    fun toBundle(): Bundle {
        val args = Bundle(1)
        args.putStringArray(ARGS_BUNDLE, arrayOf(suit, value))
        return args
    }

    override fun toString(): String {
        val instance = BidiFormatter.getInstance();
        return if (!instance.isRtlContext) {
            instance.unicodeWrap("$value $suit")
        } else {
            instance.unicodeWrap("$suit $value")
        }
    }

    companion object {
        //private val ARGS_BUNDLE = Card::class.java.name + ":Bundle"
        private val ARGS_BUNDLE = "${Card::class.java.simpleName}:Bundle"

        private val SUITS = setOf("♣" /* clubs*/, "♦" /* diamonds*/, "♥" /* hearts*/, "♠" /*spades*/)
        private val VALUES = setOf("2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A")

        val DECK = SUITS.flatMap { suit ->
            VALUES.map { value ->
                Card(suit, value)
            }
        }

        fun List<Card>.find(value: String, suit: String): Card? {
            return find { it.value == value && it.suit == suit }
        }

        /** Use in conjunction with [Card.toBundle]  */
        fun fromBundle(bundle: Bundle): Card {
            val spec = bundle.getStringArray(ARGS_BUNDLE)
            return Card(spec!![0], spec[1])
        }
    }
}